src/dex/nerve/nerve.ts

```diff
@@ -0,0 +1,405 @@
+import _ from 'lodash';
+import { bigIntify } from './utils';
+
+export class Nerve
+  extends SimpleExchange
+  implements IDex<NerveData, DexParams, OptimizedNerveData>
+{
+  protected eventPools: EventPoolMappings = {};
+
+  readonly hasConstantPriceLargeAmounts = false;
+
+  readonly minConversionRate = '1';
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(NerveConfig);
+
+  logger: Logger;
+
+  static getIdentifier(dexKey: string, poolAddress: string) {
+    return `${dexKey.toLowerCase()}_${poolAddress.toLowerCase()}`;
+  }
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+    protected poolConfigs = NerveConfig[dexKey][network].poolConfigs,
+    protected nervePoolIface = new Interface(nervePoolABIDefault),
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  get allPools() {
+    return Object.values(this.eventPools);
+  }
+
+  async initializePricing(blockNumber: number) {
+    await Promise.all(
+      Object.values(this.poolConfigs).map(poolConfig => {
+        const poolIdentifier = Nerve.getIdentifier(
+          this.dexKey,
+          poolConfig.address,
+        );
+
+        // We don't support Metapool yet
+        if (!poolConfig.isMetapool) {
+          this.eventPools[poolIdentifier] = new NerveEventPool(
+            this.dexKey,
+            this.network,
+            this.dexHelper,
+            this.logger,
+            poolConfig.name,
+          );
+          // Generate first state for the blockNumber
+          return this.eventPools[poolIdentifier].setup(blockNumber);
+        }
+      }),
+    );
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] || null;
+  }
+
+  async getStates(
+    pools?: EventPoolOrMetapool[],
+    blockNumber?: number,
+  ): Promise<DeepReadonly<{ state: PoolState; pool: EventPoolOrMetapool }[]>> {
+    const _pools = pools === undefined ? this.allPools : pools;
+
+    const _blockNumber =
+      blockNumber === undefined
+        ? await this.dexHelper.provider.getBlockNumber()
+        : blockNumber;
+
+    return Promise.all(
+      _pools.map(async eventPool => {
+        let state = eventPool.getState(_blockNumber);
+        if (!state || !state.isValid) {
+          this.logger.info(
+            `State for ${this.dexKey} pool ${eventPool.name} is stale or invalid on block ${_blockNumber}. Generating new one`,
+          );
+          const newState = await eventPool.generateState(_blockNumber);
+          eventPool.setState(newState, _blockNumber);
+          return { state: newState, pool: eventPool };
+        } else {
+          return { state, pool: eventPool };
+        }
+      }),
+    );
+  }
+
+  async getPoolIdentifiers(
```

Comment:
the `getPriceVolume` does wrapETH so the `getPoolIdentifier` should also do the same!

Reply:
Fixed